lib/ansible/cli/doc.py

```diff
@@ -116,11 +307,19 @@ def init_parser(self):
         self.parser.add_argument('args', nargs='*', help='Plugin', metavar='plugin')
         self.parser.add_argument("-t", "--type", action="store", default='module', dest='type',
                                  help='Choose which plugin type (defaults to "module"). '
-                                      'Available plugin types are : {0}'.format(C.DOCUMENTABLE_PLUGINS),
-                                 choices=C.DOCUMENTABLE_PLUGINS)
+                                      'Available plugin types are : {0}'.format(C.DOCUMENTABLE_PLUGINS + ('role',)),
+                                 choices=C.DOCUMENTABLE_PLUGINS + ('role',))
         self.parser.add_argument("-j", "--json", action="store_true", default=False, dest='json_format',
                                  help='Change output into json format.')
 
+        # role-specific options
+        self.parser.add_argument("-r", "--roles-path", dest='roles_path', default=C.DEFAULT_ROLES_PATH,
+                                 type=opt_help.unfrack_path(pathsep=True),
+                                 action=opt_help.PrependListAction,
+                                 help='The path to the directory containing your roles.')
+        self.parser.add_argument("-e", "--entry-point", dest="entry_point",
+                                 help="Select the entry point for role(s).")
```

Comment:
should `--list` filter by entry point if one is provided?

Reply:
Okay. Might want to make this option mutually exclusive?

Reply:
Resolved in latest commit.